core/schema/cache.go

```diff
@@ -19,8 +20,9 @@ func (s *cacheSchema) Name() string {
 
 func (s *cacheSchema) Install() {
 	dagql.Fields[*core.Query]{
-		dagql.Func("cacheVolume", s.cacheVolume).
+		dagql.NodeFunc("cacheVolume", s.cacheVolume).
 			Doc("Constructs a cache volume for a given cache key.").
+			Impure("cache volume should be namespaced").
```

Comment:
I think this can now be rebased to use the new dynamic purity in https://github.com/dagger/dagger/pull/8971?

Reply:
Dynamic purity was swapped out for elective purity - so now it's correct to mark this `Impure`. Instead, the `dagql.Select` call sets `Pure: true` when it re-execs itself.